compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -601,6 +601,12 @@ impl<'a> Parser<'a> {
         self.last_unexpected_token_span = Some(self.token.span);
         let mut err = self.struct_span_err(self.token.span, &msg_exp);
 
+		if let TokenKind::Ident(symbol, _) = &self.prev_token.kind {
+			if symbol.to_string() == "public" {
+				err.span_suggestion_short(self.prev_token.span, "write `pub` instead of `public` to make the object public", "pub", appl);
```

Comment:
```suggestion
				err.span_suggestion_short(self.prev_token.span, "write `pub` instead of `public` to make the item public", "pub", appl);
```
I think we tend to use "item" in cases like this rather than "object".